test/e2e/clusterctl_upgrade_test.go

```diff
@@ -58,21 +58,6 @@ var _ = Describe("When testing clusterctl upgrades (v0.4=>current)", func() {
 	})
 })
 
-var _ = Describe("When testing clusterctl upgrades (v1.2=>current)", func() {
```

Comment:
NOTE: I think we should document somewhere that step upgrades are not supported

Reply:
Absolutely fine for me. Do you have a suggestion where? Is it okay to do this in a follow-up PR given that I think it's unrelated to this PR.

I think it's debatable what sort of upgrades Cluster API / clusterctl supports. I think clusterctl allows a lot, especially when the contract is the same (not sure). But up until now nobody actually ran e2e tests for skip-level upgrades and I wouldn't want to put the burden on the community / us to start supporting it (that's why I'm dropping those tests in this PR, we didn't have them for older CAPI releases)

Reply:
No need to do this in this PR.
I think that for starting we can document this in the clusterctl update section (and maybe make sure it enforces this rule)

Reply:
Discussed with Fabrizio. Given v1.2 is still in support (in previous releases we only had one v1beta1 version in support) we decided to re-add v1.2 tests.